README.md

```diff
@@ -93,23 +117,134 @@ externalFilters?:  boolean;
 externalFiltersState?:  ColumnFiltersState;
 
 };
+
 ```
 
 **columns:** an Array of ColumnDef<T> that specify the columns with all the props needed (see Tanstack docs)
+
 **useQueryHook:** The query hook generated by your Redux Toolkit Query API for fetching table data that should return an object like this
 
-    data: {
-    	data:  retData.data,
-    	count:  retData.count,
-    	page:  page,
-    	limit:  pageSize,
-    }
+data: {
+
+data: retData.data,
+
+count: retData.count,
+
+page: page,
+
+limit: pageSize,
+
+}
 
 **isDraggable:** Enables column sorting trough drag & drop
+
 **disableHeaderFilters:** this remove all the column headers filters (if you dont want to specify all _false_ in the **columns** array
+
 **externalFilters:** this enable the use of external filters (external means that filters are not in the columns header)
+
 **externalFiltersState:** _ColumnFiltersState_ object with the state of external filters (this is mandatory if **externalFilters** is true
 
+## Authentication
+
+This project is implemented with a flexible authentication provider that can support multiple providers.
+Now it support **MSAL** and **AUTH0** providers and you can switch between one another easily.
+This is how:
+
+1.  Go to the **index.tsx** file
+2.  Instantiate the implementation of your choice of AuthProvider interface
+
+```Typescript
+const  authProvider  =  new  Auth0AuthProvider(env);
```

Comment:
what is env? where is created from?

Reply:
it's explained 2 rows below

Reply:
it's described how to configure it locally, not what shall contain to work.